// @openrouter/sdk/esm/models/operations/getbenchmarksdesignarena.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 86e27de57b33
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as openEnums from "../../types/enums.js";
/**
 * Arena to query. Defaults to `models`.
 */
export const Arena = {
    Models: "models",
    Builders: "builders",
    Agents: "agents",
};
/** @internal */
export const Arena$outboundSchema = openEnums
    .outboundSchema(Arena);
/** @internal */
export const GetBenchmarksDesignArenaRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    arena: Arena$outboundSchema.default("models"),
    category: z.string().optional(),
    maxResults: z.int().default(50),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        maxResults: "max_results",
    });
});
export function getBenchmarksDesignArenaRequestToJSON(getBenchmarksDesignArenaRequest) {
    return JSON.stringify(GetBenchmarksDesignArenaRequest$outboundSchema.parse(getBenchmarksDesignArenaRequest));
}
