// @openrouter/sdk/esm/models/operations/getmodels.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f1343fbaaed5
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as openEnums from "../../types/enums.js";
/**
 * Filter models by use case category
 */
export const GetModelsCategory = {
    Programming: "programming",
    Roleplay: "roleplay",
    Marketing: "marketing",
    MarketingSeo: "marketing/seo",
    Technology: "technology",
    Science: "science",
    Translation: "translation",
    Legal: "legal",
    Finance: "finance",
    Health: "health",
    Trivia: "trivia",
    Academia: "academia",
};
/**
 * Sort the returned models server-side. Prefer this over fetching the full list and sorting client-side. Options: pricing-low-to-high, pricing-high-to-low (average prompt/completion price), context-high-to-low (context length), throughput-high-to-low, latency-low-to-high (recent median performance), most-popular, top-weekly (tokens processed in the last week), newest (creation date). When omitted, the existing default ordering is preserved.
 */
export const GetModelsSort = {
    MostPopular: "most-popular",
    Newest: "newest",
    TopWeekly: "top-weekly",
    PricingLowToHigh: "pricing-low-to-high",
    PricingHighToLow: "pricing-high-to-low",
    ContextHighToLow: "context-high-to-low",
    ThroughputHighToLow: "throughput-high-to-low",
    LatencyLowToHigh: "latency-low-to-high",
};
/**
 * Filter by distillation capability. "true" returns only distillable models, "false" excludes them.
 */
export const Distillable = {
    True: "true",
    False: "false",
};
/**
 * When set to "true", return only models with zero data retention endpoints.
 */
export const Zdr = {
    True: "true",
};
/**
 * Filter to models with endpoints in the given data region. Currently only "eu" is supported.
 */
export const Region = {
    Eu: "eu",
};
/** @internal */
export const GetModelsCategory$outboundSchema = openEnums.outboundSchema(GetModelsCategory);
/** @internal */
export const GetModelsSort$outboundSchema = openEnums.outboundSchema(GetModelsSort);
/** @internal */
export const Distillable$outboundSchema = openEnums.outboundSchema(Distillable);
/** @internal */
export const Zdr$outboundSchema = z.enum(Zdr);
/** @internal */
export const Region$outboundSchema = z.enum(Region);
/** @internal */
export const GetModelsRequest$outboundSchema = z.object({
    httpReferer: z.string().optional(),
    appTitle: z.string().optional(),
    appCategories: z.string().optional(),
    category: GetModelsCategory$outboundSchema.optional(),
    supportedParameters: z.string().optional(),
    outputModalities: z.string().optional(),
    sort: GetModelsSort$outboundSchema.optional(),
    q: z.string().optional(),
    inputModalities: z.string().optional(),
    context: z.int().optional(),
    minPrice: z.nullable(z.number()).optional(),
    maxPrice: z.nullable(z.number()).optional(),
    arch: z.string().optional(),
    modelAuthors: z.string().optional(),
    providers: z.string().optional(),
    distillable: Distillable$outboundSchema.optional(),
    zdr: Zdr$outboundSchema.optional(),
    region: Region$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        httpReferer: "HTTP-Referer",
        supportedParameters: "supported_parameters",
        outputModalities: "output_modalities",
        inputModalities: "input_modalities",
        minPrice: "min_price",
        maxPrice: "max_price",
        modelAuthors: "model_authors",
    });
});
export function getModelsRequestToJSON(getModelsRequest) {
    return JSON.stringify(GetModelsRequest$outboundSchema.parse(getModelsRequest));
}
