// @openrouter/sdk/esm/models/benchmarksdameta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: caee78d8918e
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Data source identifier.
 */
export const BenchmarksDAMetaSource = {
    DesignArena: "design-arena",
};
/**
 * URL of the upstream data source.
 */
export const BenchmarksDAMetaSourceUrl = {
    HttpsWwwDesignarenaAi: "https://www.designarena.ai",
};
/**
 * Dataset version.
 */
export const BenchmarksDAMetaVersion = {
    V1: "v1",
};
/** @internal */
export const EloBounds$inboundSchema = z.object({
    max: z.number(),
    min: z.number(),
});
export function eloBoundsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EloBounds$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EloBounds' from JSON`);
}
/** @internal */
export const BenchmarksDAMetaSource$inboundSchema = z.enum(BenchmarksDAMetaSource);
/** @internal */
export const BenchmarksDAMetaSourceUrl$inboundSchema = z.enum(BenchmarksDAMetaSourceUrl);
/** @internal */
export const BenchmarksDAMetaVersion$inboundSchema = z.enum(BenchmarksDAMetaVersion);
/** @internal */
export const BenchmarksDAMeta$inboundSchema = z.object({
    arena: z.string(),
    as_of: z.string(),
    category: z.nullable(z.string()),
    citation: z.string(),
    elo_bounds: z.lazy(() => EloBounds$inboundSchema),
    model_count: z.int(),
    source: BenchmarksDAMetaSource$inboundSchema,
    source_url: BenchmarksDAMetaSourceUrl$inboundSchema,
    version: BenchmarksDAMetaVersion$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "as_of": "asOf",
        "elo_bounds": "eloBounds",
        "model_count": "modelCount",
        "source_url": "sourceUrl",
    });
});
export function benchmarksDAMetaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenchmarksDAMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenchmarksDAMeta' from JSON`);
}
