// @openrouter/sdk/esm/models/benchmarksaaitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 82b5c9f01e00
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { BenchmarkPricing$inboundSchema, } from "./benchmarkpricing.js";
/** @internal */
export const BenchmarksAAItem$inboundSchema = z.object({
    aa_name: z.string(),
    agentic_index: z.nullable(z.number()),
    coding_index: z.nullable(z.number()),
    intelligence_index: z.nullable(z.number()),
    model_permaslug: z.string(),
    pricing: z.nullable(BenchmarkPricing$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "aa_name": "aaName",
        "agentic_index": "agenticIndex",
        "coding_index": "codingIndex",
        "intelligence_index": "intelligenceIndex",
        "model_permaslug": "modelPermaslug",
    });
});
export function benchmarksAAItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenchmarksAAItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenchmarksAAItem' from JSON`);
}
