// @openrouter/sdk/esm/models/benchmarksaameta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ea7ac20c3f7c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Data source identifier.
 */
export const BenchmarksAAMetaSource = {
    ArtificialAnalysis: "artificial-analysis",
};
/**
 * URL of the upstream data source.
 */
export const BenchmarksAAMetaSourceUrl = {
    HttpsArtificialanalysisAi: "https://artificialanalysis.ai",
};
/**
 * Dataset version.
 */
export const BenchmarksAAMetaVersion = {
    V1: "v1",
};
/** @internal */
export const BenchmarksAAMetaSource$inboundSchema = z.enum(BenchmarksAAMetaSource);
/** @internal */
export const BenchmarksAAMetaSourceUrl$inboundSchema = z.enum(BenchmarksAAMetaSourceUrl);
/** @internal */
export const BenchmarksAAMetaVersion$inboundSchema = z.enum(BenchmarksAAMetaVersion);
/** @internal */
export const BenchmarksAAMeta$inboundSchema = z.object({
    as_of: z.string(),
    citation: z.string(),
    model_count: z.int(),
    source: BenchmarksAAMetaSource$inboundSchema,
    source_url: BenchmarksAAMetaSourceUrl$inboundSchema,
    version: BenchmarksAAMetaVersion$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "as_of": "asOf",
        "model_count": "modelCount",
        "source_url": "sourceUrl",
    });
});
export function benchmarksAAMetaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenchmarksAAMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenchmarksAAMeta' from JSON`);
}
